src/gui/sharelinkwidget.cpp

```diff
@@ -202,14 +207,16 @@ void ShareLinkWidget::setupUiOptions()
 
     if (_linkShare.data()->isPasswordSet()) {
         _passwordProtectLinkAction->setChecked(true);
-        _ui->lineEdit_password->setPlaceholderText("********");
-        showPasswordOptions(true);
+        _ui->lineEdit_password->setPlaceholderText(passwordIsSetPlaceholder);
+        showPasswordOptions(true, justCreated);
```

Comment:
I wouldn't have the second boolean parameter which breaks down the `showPasswordOptions` API. You could instead have instead this:
```
if (justCreated) {
    _ui->lineEdit_password->setFocus();
}
```

I'd go even one step further in that design if possible and remove the whole chain of "justCreated" bool parameters which got added. Just to make explicit the "this is a new one" flag with its own call higher up in the food chain, probably with something along the lines of:
```
if (thisIsANewOne) {
   markAsNew();
}
```

Reply:
@er-vin The problem with this:

```
if (thisIsANewOne) {
   markAsNew();
}
```

Is, I need this logic of "if thisIsAnewOne then do the following..." executed just once. I don't need to store a flag. If I do, I risk breaking the logic when further calls to this code are made. Then, I will also need **unMarkAsNew** - which leads to 2 methods **mark/unmark** or **setMarkedAsNew** added. So, much more code that doesn't do much work.

Also, dragging this information in the **linkShare** is not a good idea as this information is not needed anywhere else. It's needed just one time - when we create a new link share so we can focus a password lineEdit.